src/db.rs

```diff
@@ -71,6 +71,61 @@ impl DB {
         Ok(())
     }
 
+    pub fn migrate<P: AsRef<Path>>(&mut self, path: P) -> Result<()> {
+        if !self.dirs.is_empty() {
+            return Err(anyhow!(
+                "To prevent conflicts, you can only migrate from z with an empty \
+                 zoxide database!"
+            ));
+        }
+
+        let zdata = File::open(path)?;
+        let reader = BufReader::new(zdata);
+
+        for line in reader.lines() {
+            let line = line?;
```

Comment:
Instead of doing this consider writing a proper iterator chain using `filter_map` and the like

Reply:
Just so people know: I did attempt to `for_each` this, but 1) the `anyhow!` macro doesn't work inside of closures; and 2) you can't `return` from a function while inside a closure.

I did forget about `filter_map`, so I did at least include that.